index.js

```diff
@@ -48,15 +48,32 @@ var pify = module.exports = function (obj, P, opts) {
 		return opts.include ? opts.include.some(match) : !opts.exclude.some(match);
 	};
 
-	var ret = typeof obj === 'function' ? function () {
-		if (opts.excludeMain) {
-			return obj.apply(this, arguments);
-		}
+	var ret;
+	if (typeof obj === 'function') {
+		ret = function () {
+			if (opts.excludeMain) {
+				return obj.apply(this, arguments);
+			}
+
+			return processFn(obj, P, opts).apply(this, arguments);
+		};
+	} else if (opts.inherited) {
+		ret = Object.create(Object.getPrototypeOf(obj));
+	} else {
+		ret = {};
+	}
 
-		return processFn(obj, P, opts).apply(this, arguments);
-	} : {};
+	var props;
+	if (opts.inherited) {
+		props = [];
+		for (var prop in obj) {	// eslint-disable-line guard-for-in
+			props.push(prop);
+		}
+	} else {
+		props = Object.keys(obj);
+	}
```

Comment:
> Sure, that would work. Alternatively, reduce could be avoided altogether by using that imperative style in both cases (or moving the if inside the loop to combine the cases). It would probably perform better on microbenchmarks.

:+1: I don't care about the microperf improvements, but I think it would make the code more readable.